src/coreclr/jit/compiler.h

```diff
@@ -5767,7 +5767,7 @@ class Compiler
     GenTree* fgMorphCopyBlock(GenTree* tree);
     GenTree* fgMorphStoreDynBlock(GenTreeStoreDynBlk* tree);
     GenTree* fgMorphForRegisterFP(GenTree* tree);
-    GenTree* fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac = nullptr);
+    GenTree* fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optAssertionPropDone = nullptr);
```

Comment:
Should we add a top level variable `optAssertionPropDone` next to `optLocalAssertionProp` instead of sending it through multiple methods?

Reply:
That won't be fine-grained enough (in the chance that say one struct copy tree has a struct copy embedded in one of its operands).